src/client/usecase.ts

```diff
@@ -70,20 +76,20 @@ export class UseCase extends UseCaseBase {
     const boundProfileProvider = await this.bind(options);
 
     // TOOD: rewrap the errors for public consumption?
-    return boundProfileProvider.perform<TInput, TOutput>(this.name, input);
+    return await boundProfileProvider.perform<TInput, TOutput>(this.name, input);
```

Comment:
I'm afraid to ask but.. what breaks without this await?

Reply:
Nope, I was just trying to localize the concurrency bug :)